src/domain.ts

```diff
@@ -91,13 +98,60 @@ function getSlidesSettings(editor: Editor): Settings {
     ...(fontFamily && { "terminal.integrated.fontFamily": fontFamily }),
     ...(previewMarkdownFiles && {
       "slides.previewMarkdownFiles": true
-    })
+    }),
+    ...(slidesFolder && { "workbench.slidesFolder": slidesFolder })
   });
 }
 
+/*
+ *   The previous version of this method used the editor interface's
+ *   open all files method to open the files in the root folder. To
+ *   make sure that another settings file wasn't created, I had to
+ *   use a more manual process.
```

Comment:
Thanks for the comment, it helps understanding the rationale behind the code.

So, in fact, the `editor` is my own adapter. It won't create another settings file or anything. It uses VS Code API to open all files, manually as you said. I think we should change `openAllFiles()` signature to take a `Folder` parameter. And update the implementation so it actually open files that are in that folder. But I think the final change will be quite small.